pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -701,9 +701,13 @@ func (lb *ListenerBuilder) waypointRouteDestination(
 		action.Timeout = in.Timeout
 	}
 	// Use deprecated value for now as the replacement MaxStreamDuration has some regressions.
+	// TODO: check and see if the replacement has been fixed.
 	// nolint: staticcheck
 	action.MaxGrpcTimeout = action.Timeout
 
+	// return 500 for invalid backends
+	// https://github.com/kubernetes-sigs/gateway-api/blob/cea484e38e078a2c1997d8c7a62f410a1540f519/apis/v1beta1/httproute_types.go#L204
+	action.ClusterNotFoundResponseCode = route.RouteAction_INTERNAL_SERVER_ERROR
```

Comment:
Should be only if gatewaySemantics?

Reply:
Don't waypoints always have gateway semantics?

Reply:
Hmm, you can use VirtualService. Or have no explicit routes at all

Reply:
Fair point - will guard with an if